feat(editor): add deletion of selected SVG nodes

xauravww opened the pull request:
- Introduce delete button for leaf nodes in AppletLayers tree (excludes root)
- Handle Delete key press in EditorCanvas to remove selected elements
- Implement deleteSelected action in editor store with XML parsing, undo support, and JSON update

Fixes #58

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Delete non-root elements directly from the layers panel via a new context action.
* **Keyboard & Shortcuts**
  * Press Delete to remove the currently selected element; default browser behavior is prevented.
* **Improvements**
  * Deletions are recorded in undo/redo history and clear selection after removal.
* **UI**
  * Added a conditional “more” button in the layers header to access delete actions for eligible nodes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

xauravww committed:
feat(editor): add deletion of selected SVG nodes

- Introduce delete button for leaf nodes in AppletLayers tree (excludes root)
- Handle Delete key press in EditorCanvas to remove selected elements
- Implement deleteSelected action in editor store with XML parsing, undo support, and JSON update

krushndayshmookh commented on src/components/applets/AppletLayers.vue:
```diff
@@ -44,6 +54,12 @@ function onTreeSelectionChange(nodeId) {
 }
 
+const deleteNode = (id) => {
+  if (id === 'svg-root') return
```
We can also disable the button for root node.

krushndayshmookh commented on src/stores/editor-store.js:
```diff
@@ -476,6 +476,21 @@ export const useEditorStore = defineStore('editor', {
+      const before = this.xml
+      const doc = new DOMParser().parseFromString(this.xml, 'image/svg+xml')
+      const el = doc.getElementById(this.selectedId)
+      if (!el || el === doc.documentElement) return // Don't delete root SVG
```
Since we are already checking for root element earlier, this check is not needed.

xauravww committed:
feat(ui): replace delete button with dropdown menu for SVG nodes

Update AppletLayers component to use a three-dot menu icon with a popup list
containing a delete option instead of a direct delete button. Adjust conditions
to show the menu for non-root nodes excluding SVG tags, and simplify editor store
logic by removing redundant checks for root SVG deletion.

xauravww commented on src/components/applets/AppletLayers.vue:
```diff
@@ -12,6 +12,15 @@
+          icon="mdi-delete"
+          class="q-ml-auto"
+          @click.stop="deleteNode(prop.node.id)"
+        />
```
Yes I have added that

xauravww commented:
Changes done as requested

krushndayshmookh committed:
Apply suggestion from @coderabbitai[bot]

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

krushndayshmookh commented:
LGTM!

Thanks @xauravww !
